device/alpaka/include/traccc/alpaka/utils/vecmem_types.hpp

```diff
@@ -102,4 +104,13 @@ struct host_device_types<::alpaka::TagGpuSyclIntel> {
     using managed_memory_resource = ::vecmem::sycl::shared_memory_resource;
     using device_copy = ::vecmem::sycl::copy;
 };
+
+using device_memory_resource =
+    typename host_device_types<AccTag>::device_memory_resource;
+using host_memory_resource =
+    typename host_device_types<AccTag>::host_memory_resource;
+using managed_memory_resource =
+    typename host_device_types<AccTag>::managed_memory_resource;
+using device_copy = typename host_device_types<AccTag>::device_copy;
```

Comment:
🤔 Not necessarily for this PR, but I'd really like if the device libraries would not publicly depend on the device/language specific VecMem libraries. Much like how they shouldn't publicly depend on CUDA/SYCL/etc.

For this, I'd imagine that Alpaka could provide a helper class, that could provide (non)-const references to memory resources and copy objects of the appropriate types. Through their abstract vecmem interfaces. So that the "decision" about what object to instantiate, could sit all in the `traccc_alpaka` private source code. 🤔